src/history/HistoryArchive.h

```diff
@@ -96,16 +98,34 @@ struct HistoryArchiveState
     void
     serialize(Archive& ar)
     {
-        ar(CEREAL_NVP(version), CEREAL_NVP(server), CEREAL_NVP(currentLedger),
-           CEREAL_NVP(currentBuckets));
+        ar(CEREAL_NVP(version), CEREAL_NVP(server), CEREAL_NVP(currentLedger));
+        try
+        {
+            ar(CEREAL_NVP(networkPassphrase));
+        }
+        catch (...)
+        {
+            // networkPassphrase wasn't parsed.
+            // This is expected when the input file does not contain it.
+        }
+        ar(CEREAL_NVP(currentBuckets));
     }
 
     template <class Archive>
     void
     serialize(Archive& ar) const
     {
-        ar(CEREAL_NVP(version), CEREAL_NVP(server), CEREAL_NVP(currentLedger),
-           CEREAL_NVP(currentBuckets));
+        ar(CEREAL_NVP(version), CEREAL_NVP(server), CEREAL_NVP(currentLedger));
+        try
```

Comment:
this is the *writer* specialization, so we don't need this try/catch

Reply:
instead what you can do is only call `ar` on `networkPassphrase` when it's not empty (and avoid putting empty strings in the json file)

Reply:
Fixed it!